test/datasets/test_conll2000chunking.py

```diff
@@ -0,0 +1,88 @@
+import os
+import random
+import string
+import gzip
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.conll2000chunking import CoNLL2000Chunking
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "CoNLL2000Chunking")
+    temp_dataset_dir = os.path.join(base_dir, "conll2000chunking")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ["train.txt", "test.txt"]:
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        mocked_lines = mocked_data[os.path.splitext(file_name)[0]]
+        with open(txt_file, "w") as f:
+            for i in range(5):
+                label = seed % 2
+                rand_strings = [random.choice(string.ascii_letters) for i in range(seed)]
+                rand_label_1 = [random.choice(string.ascii_letters) for i in range(seed)]
+                rand_label_2 = [random.choice(string.ascii_letters) for i in range(seed)]
+                # one token per line (each sample ends with an extra \n)
+                for rand_string, label_1, label_2 in zip(rand_strings, rand_label_1, rand_label_2):
+                    f.write(f"{rand_string} {label_1} {label_2}\n")
+                f.write("\n")
+                dataset_line = (rand_strings, rand_label_1, rand_label_2)
+                # append line to correct dataset split
+                mocked_lines.append(dataset_line)
+                seed += 1
+
+
+    # create gz file from dataset folder
+    for file_name in ("train.txt", "test.txt"):
+        compressed_dataset_path = os.path.join(temp_dataset_dir, f"{file_name}.gz")
+        with gzip.open(compressed_dataset_path, "wb") as gz_file:
+            txt_file = os.path.join(temp_dataset_dir, file_name)
+            with open(txt_file, "rb") as file_in:
+                gz_file.writelines(file_in)
```

Comment:
Is there any way to simplify this? Can we add the entire folder to the archive like we've been doing for tar archives?

Reply:
I looked into it and there does not seem to be a way to use `gzip` to compress an entire folder. I simplified the logic for readability.